File: src/main/java/com/bot/commands/settings/DefaultVolumeCommand.java
package com.bot.commands.settings;

import com.bot.commands.ModerationCommand;
import com.bot.db.GuildDAO;
import com.bot.utils.CommandPermissions;
import com.jagrosh.jdautilities.command.CommandEvent;

import java.util.logging.Logger;

public class DefaultVolumeCommand extends ModerationCommand {

    private static final Logger LOGGER = Logger.getLogger(com.bot.commands.voice.VolumeCommand.class.getName());
    private GuildDAO guildDAO;

    public DefaultVolumeCommand() {
        this.name = "dvolume";
        this.arguments = "<Volume 1-200>";
        this.help = "Sets the default volume for the server";
        this.guildDAO = GuildDAO.getInstance();
    }

    @Override
    protected void execute(CommandEvent commandEvent) {
        metricsManager.markCommand(this, commandEvent.getAuthor(), commandEvent.getGuild());
        // Check the permissions to do the command
        if (!CommandPermissions.canExecuteCommand(this, commandEvent))
            return;

        int newVolume;
        try{
            newVolume = Integer.parseInt(commandEvent.getArgs().split(" ")[0]);
            if (newVolume > 200 || newVolume < 0) {
                throw new NumberFormatException();
            }

            if (!guildDAO.updateGuildVolume(commandEvent.getGuild().getId(), newVolume)) {
                commandEvent.reply(commandEvent.getClient().getError() + " Something went wrong updating the default volume.");
                return;
            }

            commandEvent.getMessage().addReaction(commandEvent.getClient().getSuccess()).queue();
        }
        catch (NumberFormatException e) {
            commandEvent.reply(commandEvent.getClient().getError() + " You must enter a volume between 0 and 200");
        }
    }

}

File: src/main/java/com/bot/commands/traditional/voice/DefaultVolumeCommand.kt
package com.bot.commands.traditional.voice

import com.bot.commands.traditional.ModerationCommand
import com.bot.voice.GuildVoiceProvider
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class DefaultVolumeCommand : ModerationCommand() {

    init {
        name = "dvolume"
        arguments = "<Volume 1-200>"
        help = "Sets the default volume for the server"
    }

    @Trace(operationName = "executeCommand", resourceName = "dvolume")
    override fun executeCommand(commandEvent: CommandEvent) {
        val newVolume: Int
        try {
            newVolume = commandEvent.args.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0].toInt()
            if (newVolume > 200 || newVolume < 0) {
                throw NumberFormatException()
            }
            if (!guildDAO.updateGuildVolume(commandEvent.guild.id, newVolume)) {
                commandEvent.reply(commandEvent.client.error + " Something went wrong updating the default volume.")
                metricsManager.markCommandFailed(this, commandEvent.author, commandEvent.guild)
                return
            }
            GuildVoiceProvider.getInstance().getGuildVoiceConnection(commandEvent.guild.idLong)?.setVolume(newVolume)
            commandEvent.reactSuccess()
        } catch (e: NumberFormatException) {
            commandEvent.reply(commandEvent.client.warning + " You must enter a volume between 0 and 200")
        }
    }
}